Fix typo in input placeholder color class

acelaya opened the pull request:
Part of https://github.com/hypothesis/client/issues/6866

This PR does two things:

1. Fix two typos in `Input` tailwind classes around placeholder text color.
2. Consolidate `Input` and `Textarea` ~text color and~ placeholder color, following the changes in https://github.com/hypothesis/client/pull/6966

~Inputs were previously inheriting the text color from the page, having inconsistent values. Now we settle to `grey-8`~

The previous placeholder color (even though it was not working due to the typo mentioned above) was too light in contrast with white background. Now we settle to `grey-6`, which has more than 4.5 contrast, as required by accessibility guidelines.

![image](https://github.com/user-attachments/assets/a6a7f019-b04a-4cde-ad3e-22a749c2e50a)
![image](https://github.com/user-attachments/assets/4fc1e37d-00f1-46b2-9cf9-2cd9b412198e)
![image](https://github.com/user-attachments/assets/fa1b964e-447f-49e9-8d45-834426d7f730)

robertknight commented:
> Inputs were previously inheriting the text color from the page, having inconsistent values. Now we settle to grey-8.

Are there any cases where we _do_ want to inherit the color that you can think of? For example input fields that set the text to red if there is an error or something. Do we have a mechanism to do that if needed?

acelaya commented:
> > Inputs were previously inheriting the text color from the page, having inconsistent values. Now we settle to grey-8.
> 
> Are there any cases where we _do_ want to inherit the color that you can think of? For example input fields that set the text to red if there is an error or something. Do we have a mechanism to do that if needed?

Our existing `feedback` capability only changes the border color ATM.

![image](https://github.com/user-attachments/assets/7e7f9549-62a2-4a56-9185-2e22ff61f95c)

I can't think on any case where we currently need to change the text color, but I think it would be reasonable to just set the placeholder text and continue inheriting the main text color.

I would in that case do the same in https://github.com/hypothesis/client/pull/6966, for consistency.

robertknight commented:
> So we could just keep grey-9 for all, via inheriting the page color.

This makes sense as an incremental step. In future we should consider whether we want to harmonize the text color between the client, LMS and other apps.

acelaya commented on src/components/input/Input.tsx:
```diff
@@ -16,7 +16,7 @@ export function inputStyles({ classes, feedback }: InputStylesOptions) {
     'focus-visible-ring ring-inset border rounded w-full p-2',
     'bg-grey-0 focus:bg-white disabled:bg-grey-1',
-    'placeholder:text-color-grey-5 disabled:placeholder:color-grey-6',
+    'placeholder:text-grey-6 disabled:placeholder:text-grey-7',
```
Neither `text-color-grey-5` or `color-grey-6` were valid tailwind classes, but even if they were, `grey-5` is too light in contrast with white.

robertknight replied:
We really ought to have a lint to check validity of classes. I suspect there was confusion here because we do have some custom [`text-color-*` properties](https://github.com/hypothesis/frontend-shared/blob/9f84354fdd7541e93664c41422b6b0989fca3aad/src/tailwind.preset.js#L67).

acelaya replied:
> I suspect there was confusion here because we do have some custom [text-color-* properties](https://github.com/hypothesis/frontend-shared/blob/9f84354fdd7541e93664c41422b6b0989fca3aad/src/tailwind.preset.js#L67).
yep, that was also my thinking.

acelaya commented:
> > So we could just keep grey-9 for all, via inheriting the page color.
> 
> This makes sense as an incremental step. In future we should consider whether we want to harmonize the text color between the client, LMS and other apps.

@robertknight can you take another look? I have reverted the `text-grey-8` class, and just addressed the typos and too low contrast for the placeholder color.

robertknight reviewed (approved):
This change seems like the most straightforward way of addressing the accessibility hazard with placeholder text. It does however mean that contrast is a weaker cue than before that text in an input field is a placeholder. This raises the importance of adding other cues such as adding an ellipsis at the end.

GitHub recently made a similar change across their input fields and it runs into the same challenges.
